Tests/CSharp/SpikeTests.cs

```diff
@@ -0,0 +1,145 @@
+
+        <main aria-labelledby="title">
+            <h2 id="title">@ViewData("Title").</h2><h3 id="title">@ViewData("Title2").</h3>
+            <h3>@ViewData("Message")</h3>
+
+            <p>Use this area to provide additional information.</p>
+        </main>@Code
+            ViewData("Title") = "About" & CStr(1)
+        End Code
+
+        <main aria-labelledby="title">
+            <h2 id="title">@ViewData("Title").</h2><h3 id="title">@ViewData("Title2").</h3>
+            <h3>@ViewData("Message")</h3>
+
+            <p>Use this area to provide additional information.</p>
+        </main>
+
+        """, "anything.vbhtml");
+        
+        var syntaxTree = codeDocument.GetSyntaxTree();
+        var getRoot = syntaxTree.GetType().GetProperty("Root", BindingFlags.Instance | BindingFlags.NonPublic).GetMethod;
+        var root = getRoot.Invoke(syntaxTree, BindingFlags.Instance | BindingFlags.NonPublic, null, null, null);
+        var getDocument = root.GetType().GetProperty("Document", BindingFlags.Instance | BindingFlags.Public).GetMethod;
+        var document = getDocument.Invoke(root, BindingFlags.Instance | BindingFlags.NonPublic, null, null, null);
+        var getChildren = document.GetType().GetProperty("Children", BindingFlags.Instance | BindingFlags.Public).GetMethod;
+        var children = getChildren.Invoke(document, BindingFlags.Instance | BindingFlags.NonPublic, null, null, null);
+
+
+    }
+
+    [UnsafeAccessor(UnsafeAccessorKind.Method)]
+    private static extern object get_Root(RazorSyntaxTree codeDocument);
+
+
+    private RazorCodeDocument GetCodeDocument(string source, string filePath)
+    {
+        var taghelper = TagHelperDescriptorBuilder.Create("TestTagHelper", "TestAssembly")
+            //.BoundAttributeDescriptor(attr => attr.Name("show").TypeName("System.Boolean"))
+            //.BoundAttributeDescriptor(attr => attr.Name("id").TypeName("System.Int32"))
+            //.TagMatchingRuleDescriptor(rule => rule.RequireTagName("taghelper"))
+            //.Metadata(TypeName("TestTagHelper"))
+            .Build();
+
+        var engine = CreateProjectEngine(_ => {
+                //return builder.Features.Add(new VisualStudioEnableTagHelpersFeature());
+            }
+        );
+
+        var sourceDocument = TestRazorSourceDocument.Create(source, normalizeNewLines: true);
+        var importDocument = TestRazorSourceDocument.Create("@addTagHelper *, TestAssembly", filePath: filePath, relativePath: filePath);
+
+        var codeDocument = engine.ProcessDesignTime(sourceDocument, FileKinds.Legacy, importSources: ImmutableArray.Create(importDocument), new[] { taghelper });
+        return codeDocument;
+        //return RazorWrapperFactory.WrapCodeDocument(codeDocument);
+    }
+    protected RazorProjectEngine CreateProjectEngine(Action<RazorProjectEngineBuilder> configure)
+    {
+        return RazorProjectEngine.Create(RazorConfiguration.Default, RazorProjectFileSystem.Create("c:\\tmp"), configure);
```

Comment:
May need an older version of the library to parse vbhtml, or just to tweak the settings here. If it's an older version, would need to check we're allowed to redistribute it, but it would solve any brittleness issues since it won't change.